services/api/src/resources/fact/resolvers.ts

```diff
@@ -186,6 +188,11 @@ export const addFact: ResolverFn = async (
   { input: { id, environment: environmentId, name, value, source, description, type, category, keyFact, service } },
   { sqlClientPool, hasPermission, userActivityLogger }
 ) => {
+  if (environmentId == undefined) {
+    logger.error(`No environment ID given for fact: ${name}`)
+    throw new Error(`No environment ID given for fact: ${name}`)
+  }
+
```

Comment:
Needed to add this since I had to remove the 'required' environmentID check from the schema input and handle that here instead.